### PR TITLE
Update README and add crate specific ones

### LINKED ISSUE #174: Add crate-specific READMEs
#### Summary
Currently only the workspace has a README. This should be true for teach crate - `piecrust` and `piecrust-uplink` - as well.

#### Possible solution design or implementation
We could just symlink the main README in the crates themselves.

#### Additional context
Crates.io looks better when a crate has a README

### PR DESCRIPTION
**Pull request by HDauven**
Resolves #174

### CODE REVIEW

**Comment by HDauven** [comment]
@ureeves I think it's ready for review now. There's maybe some capitalization issues here and there. And I used `smart contract`, `contract` and `module` interchangeably before. It's mostly standardized to `module` now.

For the `CONTRIBUTE` file I point towards an external repo. Not sure if that's nice or not. We might be able to just point to the community health files, but I'm not sure if the `.github` currently overwrites the organization wide structure.

File: piecrust-uplink/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+[![Repository](https://img.shields.io/badge/github-piecrust-blueviolet?logo=github)](https://github.com/dusk-network/piecrust)
+![Build Status](https://github.com/dusk-network/piecrust/workflows/build/badge.svg)
+[![Documentation](https://img.shields.io/badge/docs-piecrust-blue?logo=rust)](https://docs.rs/piecrust-uplink/)
+
+Piecrust Uplink is the library that allows you to build smart contracts directly on top of Dusk's Piecrust virtual machine. 
+
+## Usage
+
+`Piecrust::Uplink` allows users of the contract platform to manage the interface and state with the host environment of the modules. The example below describes a barebones module. For more detailed examples, see the [modules](https://github.com/dusk-network/piecrust/tree/main/modules) folder.
```
Comment (ureeves) [inline piecrust-uplink/README.md:11]:
```suggestion
The library allows users of the contract platform to manage the interface and state with the host environment of the modules. The example below describes a barebones module. For more detailed examples, see the [modules](https://github.com/dusk-network/piecrust/tree/main/modules) folder.
```

**Review by ureeves — APPROVED** [review]
Very nice. One small nit otherwise looks fine. Referring to an external repo is ok as well, especially since it's ours.

File: piecrust-uplink/README.md (hunk 1 above)
Comment (HDauven) [reply piecrust-uplink/README.md]: > I'm not sure this is the way to go, but at least the stylization is a little weird.

Agreed. Used your suggestion. It should suffice for now.

**Pull request merged**